### PR TITLE
feat: add cwgo docs

### PR DESCRIPTION
**Pull request by L2ncE**
#### What type of PR is this?

feat

#### What this PR does / why we need it (en: English/zh: Chinese):

en: Add cwgo docs.
zh: 新增 cwgo 文档。

#### Which issue(s) this PR fixes:

NA

### CODE REVIEW

**Comment by li-jin-gou** [comment]
PATL @Duslia  @welkeyever @GuangmingLuo

**Review by li-jin-gou — APPROVED** [review]

**Review by Duslia — APPROVED** [review]

File: content/en/docs/cwgo/getting-started/_index.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+```shell
+$ mkdir -p $(go env GOPATH)/src/github.com/cloudwego
+$ cd $(go env GOPATH)/src/github.com/cloudwego
+```
+
+2. If you put the code outside GOPATH, you can get it directly
+
+## Precautions
+
+The bottom layer of cwgo uses [kitex](https://www.cloudwego.io/zh/docs/kitex/tutorials/code-gen/code_generation/), [hz](https://www.cloudwego.io/zh/docs/hertz/tutorials/toolkit/toolkit/), [gen](https://gorm.io/gen/index.html) tools, so the corresponding tool specifications also need to be followed, such as [kitex precautions](https://www.cloudwego.io/zh/docs/kitex/tutorials/code-gen/code_generation/#%E4%BD%BF%E7%94%A8-protobuf-idl-%E7%9A%84%E6%B3%A8%E6%84%8F%E4%BA%8B%E9%A1%B9) and [Notes for hz](https://www.cloudwego.io/zh/docs/hertz/tutorials/toolkit/toolkit/#%E6%B3%A8%E6%84%8F%E4%BA%8B%E9%A1%B9).
```
Comment (GuangmingLuo) [inline content/en/docs/cwgo/getting-started/_index.md:88]: Please use the en version of the doc:
1. https://www.cloudwego.io/docs/kitex/tutorials/code-gen/code_generation/
2. https://www.cloudwego.io/docs/hertz/tutorials/toolkit/toolkit/#notes

File: content/en/docs/cwgo/getting-started/_index.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+```
+
+6. Initiate the call
+
+```shell
+$ curl http://127.0.0.1:8080/ping
+pong
+```
+
+congratulations! So far you have successfully written a Cwgo server and completed a call!
```
Comment (GuangmingLuo) [inline content/en/docs/cwgo/getting-started/_index.md:154]: -> Congratulations

File: content/en/docs/cwgo/tutorials/cli/_index.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+  | HTTP server/client
+  |-----------> hz
+  | RPC server/client
+  |-----------> kitex
+  | DB
+  ------------> gorm/gen
+ 
+```
+
+Therefore, the precautions of corresponding tools also need to be followed, such as [kitex precautions](https://www.cloudwego.io/zh/docs/kitex/tutorials/code-gen/code_generation/#%E4%BD%BF%E7%94%A8-protobuf-idl-%E7%9A%84%E6%B3%A8%E6%84%8F%E4%BA%8B%E9%A1%B9) and generate HTTP code [Notes on hz](https://www.cloudwego.io/zh/docs/hertz/tutorials/toolkit/toolkit/#%E6%B3%A8%E6%84%8F%E4%BA%8B%E9%A1%B9).
```
Comment (GuangmingLuo) [inline content/en/docs/cwgo/tutorials/cli/_index.md:27]: please use en version

File: content/en/docs/cwgo/tutorials/templete-extension/_index.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+     key: Test{{.Name}} # The searched function name, if the rendered function name exists, it is considered that the method does not need to be appended
+     append_tpl: # updated content template
+     import_tpl: # The newly added import content is a list, which can be rendered by template
+loop_method: true # Whether to enable loop rendering
+body: template content # template content
+```
+
+3. The data used by the template is PackageInfo. It is considered that this part contains all metadata, such as methodInfo, etc. The user only needs to pass the template file, and the data in the template is PackageInfo data. Commonly used content in PackageInfo can be found in the appendix.
+3. cwgo supports circularly rendering files according to methodinfo. There is only one element in the methodInfo list during cyclic rendering, which is the method currently being rendered.
+3. When updating, currently supports overwriting, skipping and adding files according to methods, and supports appending in one file. If the loop render file is enabled, only skip and cover are supported.
```
Comment (GuangmingLuo) [inline content/en/docs/cwgo/tutorials/templete-extension/_index.md:38]: 这里的几个序号还是按序递增吧

File: content/zh/docs/cwgo/tutorials/db/_index.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
+---
+title: "DB"
+linkTitle: "DB"
+weight: 4
+description: >
+
+---
+
+cwgo 集成了 gorm/gen 用于帮助用户生成 Model 代码以及基础的CURD代码。
```
Comment (GuangmingLuo) [inline content/zh/docs/cwgo/tutorials/db/_index.md:9]: CURD + 空格

Comment (L2ncE) [reply content/zh/docs/cwgo/tutorials/db/_index.md]: all fixed, thx~

File: content/en/docs/cwgo/getting-started/_index.md (hunk 1 above)
Comment (L2ncE) [reply content/en/docs/cwgo/getting-started/_index.md]: done

File: content/en/docs/cwgo/tutorials/cli/_index.md (hunk 3 above)
Comment (L2ncE) [reply content/en/docs/cwgo/tutorials/cli/_index.md]: done

File: content/en/docs/cwgo/getting-started/_index.md [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+```shell
+$ mkdir -p $(go env GOPATH)/src/github.com/cloudwego
+$ cd $(go env GOPATH)/src/github.com/cloudwego
+```
+
+2. If you put the code outside GOPATH, you can get it directly
+
+## Precautions
+
+The bottom layer of cwgo uses [kitex](https://www.cloudwego.io/docs/kitex/tutorials/code-gen/code_generation/), [hz](https://www.cloudwego.io/docs/hertz/tutorials/toolkit/toolkit/), [gen](https://gorm.io/gen/index.html) tools, so the corresponding tool specifications also need to be followed, such as [kitex precautions](https://www.cloudwego.io/docs/kitex/tutorials/code-gen/code_generation/#%E4%BD%BF%E7%94%A8-protobuf-idl-%E7%9A%84%E6%B3%A8%E6%84%8F%E4%BA%8B%E9%A1%B9) and [Notes for hz](https://www.cloudwego.io/docs/hertz/tutorials/toolkit/toolkit/#%E6%B3%A8%E6%84%8F%E4%BA%8B%E9%A1%B9).
```
Comment (GuangmingLuo) [inline content/en/docs/cwgo/getting-started/_index.md:88]: 这里不对啊，不是把 zh 删除就可以了，你这有二级标题的 tag，英文和中文不一样的。
1. https://www.cloudwego.io/docs/kitex/tutorials/code-gen/code_generation/#notes-for-using-protobuf-idls
2. https://www.cloudwego.io/docs/hertz/tutorials/toolkit/toolkit/#notes

File: content/en/docs/cwgo/tutorials/cli/_index.md (hunk 3 above)
Comment (li-jin-gou) [reply content/en/docs/cwgo/tutorials/cli/_index.md]: done

File: content/en/docs/cwgo/getting-started/_index.md (hunk 1 above)
Comment (li-jin-gou) [reply content/en/docs/cwgo/getting-started/_index.md]: done

File: content/en/docs/cwgo/getting-started/_index.md (hunk 6 above)
Comment (li-jin-gou) [reply content/en/docs/cwgo/getting-started/_index.md]: done

**Review by GuangmingLuo — APPROVED** [review]

**Review by li-jin-gou — COMMENTED** [review]
LGTM

**Review by li-jin-gou — APPROVED** [review]

**Pull request merged**